src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py

```diff
@@ -1244,3 +1244,18 @@ def save(
             sample=sample,
             **kwargs,
         )
+
+    def _update_signatures(self, code: Path, data: dict) -> bool:
+        """Update signatures for flex flow. Raise validation error if signature is not valid."""
+        if not is_flex_flow(yaml_dict=data):
+            return False
+        entry = data.get("entry")
+        signatures, _, _ = self._infer_signature(entry=entry, code=code)
+        merged_signatures = self._merge_signature(extracted=signatures, signature_overrides=data)
+        FlexFlow(path=code / FLOW_FLEX_YAML, code=code, data=data, entry=entry)._validate()
+        updated = False
+        for field in ["inputs", "outputs", "init"]:
+            if merged_signatures.get(field) != data.get(field):
```

Comment:
possible that merged_signatures doesn't have `init` or `outputs`?

Reply:
I think ff merged signatures don't have `init`, then data won't have `init`? Otherwise `init` will exist in merged signatures?

Reply:
Offline synced, dict.get won't raise error when key not exist.